client/components/site-selector/index.jsx

```diff
@@ -291,8 +314,11 @@ const SiteSelector = React.createClass( {
 } );
 
 export default connect( ( state ) => {
+	const visibleSiteCount = get( getCurrentUser( state ), 'visible_site_count', 0 );
 	return {
 		showRecentSites: get( getCurrentUser( state ), 'visible_site_count', 0 ) > 11,
-		recentSites: getPreference( state, 'recentSites' )
+		recentSites: getPreference( state, 'recentSites' ),
+		siteCount: get( getCurrentUser( state ), 'site_count', 0 ),
+		visibleSiteCount: visibleSiteCount,
```

Comment:
You could use [object property shorthand](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#New_notations_in_ECMAScript_2015) here (there's even an [ESLint rule](http://eslint.org/docs/rules/object-shorthand) if we care to standardize its usage).